pgmpy/models/DiscreteBayesianNetwork.py

```diff
@@ -1331,8 +1358,27 @@ def simulate(
 
         # Step 1: If do or virtual_intervention is specified, modify the network structure.
         if (do != {}) or (virtual_intervention != []):
-            virt_nodes = [cpd.variables[0] for cpd in virtual_intervention]
-            model = model.do(list(do.keys()) + virt_nodes)
+            # Create a combined intervention dictionary for hard interventions
+            combined_interventions = do.copy()
```

Comment:
`combined_interventions` seems a confusing name to me. Maybe `hard_intervention` or just keep it as `do` ?